src/azure-cli-core/azure/cli/core/extension/__init__.py

```diff
@@ -101,22 +101,18 @@ def preview(self):
         """
         try:
             if not isinstance(self._preview, bool):
-                self._preview = bool(self.metadata.get(EXT_METADATA_ISPREVIEW))
+                self._preview = (bool(self.metadata.get(EXT_METADATA_ISPREVIEW)) or
+                                 bool(self.metadata.get(EXT_METADATA_ISEXPERIMENTAL)) or
+                                 is_preview_from_semantic_version(self._version))
         except Exception:  # pylint: disable=broad-except
             logger.debug("Unable to get extension preview status: %s", traceback.format_exc())
         return self._preview
 
     @property
     def experimental(self):
         """
-        Lazy load experimental status.
-        Returns the experimental status of the extension.
+        In extension semantic versioning, experimental = preview, experimental deprecated
         """
-        try:
-            if not isinstance(self._experimental, bool):
-                self._experimental = bool(self.metadata.get(EXT_METADATA_ISEXPERIMENTAL))
-        except Exception:  # pylint: disable=broad-except
-            logger.debug("Unable to get extension experimental status: %s", traceback.format_exc())
         return self._experimental
```

Comment:
Why still return self._experimental?

Reply:
Experimental is justified as preview. Disable lazy load experimental, return None when needed

Reply:
Maybe better to remove this function as it always returns None